astropy/table/operations.py

```diff
@@ -37,6 +47,13 @@
 __doctest_requires__ = {"join_skycoord": ["scipy"], "join_distance": ["scipy"]}
 
 
+class _JoinSortBy(enum.Enum):
+    # TODO: if directly exposed, use an StrEnum in not PYTHON_LT_311
+    LEFT = enum.auto()
+    RIGHT = enum.auto()
+    KEYS = enum.auto()
```

Comment:
Is this class related to typing? What does it enable? (Just curious for my education!)

Reply:
Enums plays very nicely with type-checkers indeed (in particular to leverage `typing.assert_never`), but I find that they are already useful to prevent mistakes without type-checkers (provided the user code is covered by tests), because a typo results in an exception being raised (`AttributeError`), while using plain strings doesn't provide this guarantee. Strings are convenient as user-provided values but I tend to replace them with enum members as soon as possible internally, so any mistake is quickly caught.